### PR TITLE
Add the ability to capture the state of a slice.

### PR DESCRIPTION
**Pull request by rsheeter**
Intended use is to capture slices from .glyphs files so we can see things like which glyphs have changed since the prior compile (wip of this in #21)

### CODE REVIEW

File: fontir/src/stateset.rs [hunk 1]
```diff
@@ -79,23 +103,31 @@ impl FileStateSet {
...
+                    .insert(new_path.clone(), State::File(FileState::of(&new_path)?));
             }
         }
         Ok(())
     }
 
+    /// Pay attention to slice, we'd quite like to know if it changes
+    ///
+    /// Path is used only as an identifier, invent any scheme you like.
+    /// my_file/my_subsection say.
```
Comment (anthrotype) [inline fontir/src/stateset.rs:124]: > Path is used only as an identifier

then why do you use `Path` and not simply a string?

Comment (anthrotype) [reply fontir/src/stateset.rs:124]: it doesn't seem like you're using any Path-specific methods for the SliceSet. 
Path/PathBuf are wrappers around OsString/OsStr so on windows they will use backslashes and do other win-specific things. Any methods that attempts to access the underlying filesystem is bound to fai..
It feels weird to me. If there was something like python's pathllib.PurePath maybe.. but it doesn't seem to be the case in rust.
I think it'd be clearer if SliceSet used a String for the entries key.

File: fontir/src/stateset.rs [hunk 2]
```diff
@@ -345,21 +400,25 @@ mod tests {
...
         assert_eq!(fs, restored);
     }
 
     #[test]
     fn read_write_bincode() {
         let temp_dir = tempdir().unwrap();
 
-        let (_, _, fs) = one_changed_file_one_not(&temp_dir);
+        let (_, _, mut fs) = one_changed_file_one_not(&temp_dir);
+        fs.track_slice(Path::new("file.glyphs/glyph/glyph_name"), "Hi World!")
```
Comment (anthrotype) [inline fontir/src/stateset.rs:417]: what I think would be clearer and more explicit as type of the key in a SliceSet is a tuple/struct where one field is an actual Path or PathBuf (i.e. the path to the file containing the slice) and a second field contains a vector of strings defining the "path" components to identify the given slice

from `Path::new("file.glyphs/glyph/glyph_name"` to `(Path::new("file.glyphs"), vec!["glyph", "glyph_name"])`

Comment (anthrotype) [reply fontir/src/stateset.rs:417]: or as a struct:

```rust
let glyph_slice_path = { path: Path::new("file.glyphs"), keys: vec!["glyph", "glyph_name"] };
```

Comment (rsheeter) [reply fontir/src/stateset.rs:417]: My thought was that path captures that the user is intended to use a "pathlike" construct as the key, e.g. even for slices you probably want a structured key. I suppose I could hide that, and track_slice could input a String.

File: fontir/src/stateset.rs (hunk 1 above)
Comment (rsheeter) [reply fontir/src/stateset.rs:124]: I view Path/PathBuf as equivalent to Python's Path, is that wrong?

Comment (anthrotype) [reply fontir/src/stateset.rs:124]: yes. pathlib.Path is same as rust's Path/PathBuf, but it seems like there's no equivalent in Rust for a [PurePath](https://docs.python.org/3/library/pathlib.html#pure-paths), ie. "objects provide path-handling operations which don’t actually access a filesystem."

Comment (rsheeter) [reply fontir/src/stateset.rs]: A Path you never use for file IO is?

But I think you are right, this doesn't need to be a Path. I made it a String.

File: fontir/src/stateset.rs (hunk 2 above)
Comment (rsheeter) [reply fontir/src/stateset.rs]: track_slice now takes &str identifiers

File: fontir/src/stateset.rs (hunk 1 above)
Comment (anthrotype) [reply fontir/src/stateset.rs]: > A Path you never use for file IO is?

it's a string that has additional path-like methods to compose, split, etc. like a path, very convenient at times.

**Comment by rsheeter** [comment]
Merging per IM discussion

**Pull request merged**